DesignSafe-CI/portal
# bug/WP-683: Fix setfacl race condition

## Overview: ##

There was a race condition present in DS onboarding where, if a user logged in but didn't exist as a user on `cloud.data.tacc.utexas.edu`, they would not have accurate ACLs set on their My Data dir. Then, if they logged in again, the previous check would use the service account to make a files listing and so would not catch the missing ACLs.

This PR changes the check to make the file listing with the user's account, and handles both the `ForbiddenError` and the `NotFoundError` responses from tapis during login checks.

## PR Status: ##


## Related Jira tickets: ##

* [WP-683](https://tacc-main.atlassian.net/browse/WP-683)

## Summary of Changes: ##
- Make check with user client during login checks, handle `ForbiddenError`
- Remove `AGAVE_WORKING_SYSTEM` from `TAPIS_SYSTEMS_TO_CONFIGURE` as is no longer used
- Formatting

## Testing Steps: ##
1. Make sure `TAPIS_TG458981_JWT` is set in your `designsafe.env` locally
2. On the cloud.data VM via the `tg458981` account, remove your user from your My Data dir at `/data/designsafe/mydata/$username`
  - e.g. `$ setfacl -x d:u:sal_test,u:sal_test /data/designsafe/mydata/sal_test`
4. Make a fresh login to the portal with this branch checked out
5. Ensure that ACLs have been fixed.

You will see this in logs
```
des_workers        | [DJANGO] $(processName)s INFO 2024-09-23 16:04:52,229 tasks designsafe.apps.auth.tasks.check_or_configure_system_and_user_directory:56: Unable to list system/files: Checked and there is a need to configure system:designsafe.storage.default path:sal_test for sal_test: message: FILES_CLIENT_SSH_PERM_DENIED OboTenant: designsafe OboUser: sal_test Operation: ls System: designsafe.storage.default EffectiveUser: sal_test Host: cloud.data.tacc.utexas.edu Path: sal_test Error: SFTP error (SSH_FX_PERMISSION_DENIED): Permission denied

...

des_workers        | [DJANGO] INFO 2024-09-23 16:04:52,399 tasks designsafe.apps.auth.tasks.check_or_configure_system_and_user_directory:71: Ensuring directory exists for user=sal_test then going to run setfacl on system=designsafe.storage.default path=sal_test
```

Then, you can check your My Data listing works, and check via `getfacl`
```
[tg458981@c4-cloud ~]$ getfacl /data/designsafe/mydata/sal_test
getfacl: Removing leading '/' from absolute path names
# file: data/designsafe/mydata/sal_test
# owner: tg458981
# group: G-816877
user::rwx
user:tg458981:rwx
user:sal_test:rwx
group::r-x
mask::rwx
other::---
default:user::rwx
default:user:tg458981:rwx
default:user:sal_test:rwx
default:group::r-x
default:mask::rwx
default:other::---
```
